File: app/src/main/java/com/example/aleksei/reminderapp/model/NoteDao.java
package com.example.aleksei.reminderapp.model;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

import io.reactivex.Single;

@Dao
public interface NoteDao {

    @Insert
    void insert(Note entity);

    @Query("SELECT * FROM Note ORDER BY noteDate")
    Single<List<Note>> getAllNotes();

    @Delete
    void delete(Note note);
}

File: app/src/main/java/com/reminderapp/mvp/data/NoteDao.kt
package com.aleksei.reminderapp.mvp.data

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import io.reactivex.Single

@Dao
interface NoteDao {
    @Query("SELECT * FROM Note ORDER BY noteDate")
    fun allNotes(): Single<List<Note>>

    @Insert
    fun insert(entity: Note)

    @Delete
    fun delete(note: Note)
}